inference-engine/src/transformations/src/transformations/snippets/assign_registers_pass.cpp

```diff
@@ -0,0 +1,247 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/snippets/assign_registers_pass.hpp"
+#include "transformations/snippets/remarks.hpp"
+#include "transformations/rt_info/register_info.hpp"
+
+#include <ngraph/opsets/opset1.hpp>
+
+#include "ngraph_ops/load.hpp"
+#include "ngraph_ops/scalar.hpp"
+#include "ngraph_ops/broadcastload.hpp"
+
+#include <iterator>
+
+// Assigning internal `vector` register indexes to Ops (virtual registers)
+// After this pass changing order of variables or datafrow lead to invalidation of register assignment
+bool ngraph::pass::AssignRegistersPass::run_on_function(std::shared_ptr<Function> f) {
+    using Reg = size_t;
```

Comment:
Please add CC macros to all net transformations.

Reply:
@ilyachur, what is CC macros? could you please point me to example? I'm not quite aware

Reply:
`RUN_ON_FUNCTION_SCOPE` macro. You can find an example here: https://github.com/openvinotoolkit/openvino/blob/e82018221c674774223d7ea9585f210791ab7f2d/inference-engine/src/transformations/src/transformations/smart_reshape/mimic_set_batch_size.cpp#L12